File: sdk/go/conformance_test.go
package sdk

import (
	"os"
	"testing"
)

// TestRealDaemonConformance is opt-in so SDK releases can run the same
// read-only smoke contract against a packaged daemon.
func TestRealDaemonConformance(t *testing.T) {
	socket := os.Getenv("CARINA_CONFORMANCE_SOCKET")
	if socket == "" {
		t.Skip("CARINA_CONFORMANCE_SOCKET is not set")
	}
	c, err := DialPath(socket)
	if err != nil {
		t.Fatal(err)
	}
	defer c.Close()
	info, err := c.Initialize("carina-sdk-go", "0.6.3")
	if err != nil {
		t.Fatal(err)
	}
	if info.ProtocolVersion == "" {
		t.Fatal("daemon omitted protocol version")
	}
	if _, err = c.Doctor(); err != nil {
		t.Fatal(err)
	}
	if _, err = c.ListAgents(""); err != nil {
		t.Fatal(err)
	}
	if _, err = c.ListWorkers(); err != nil {
		t.Fatal(err)
	}
	if _, err = c.ListWorkflows(); err != nil {
		t.Fatal(err)
	}
	if _, err = c.ListExtensions(); err != nil {
		t.Fatal(err)
	}
}

File: sdk/typescript/test/conformance.test.mjs
import assert from 'node:assert/strict'
import test from 'node:test'
import { CarinaClient } from '../dist/index.js'

const socket = process.env.CARINA_CONFORMANCE_SOCKET
test('packaged daemon read-only conformance', { skip: !socket }, async () => {
  const client = new CarinaClient(socket, 5_000)
  try {
    const runtime = await client.initialize()
    assert.match(runtime.runtime_version, /^\d+\.\d+\.\d+$/)
    assert.equal(runtime.protocol_version.split('.')[0], '1')
    await client.doctor(); await client.listAgents(); await client.listWorkers(); await client.listWorkflows(); await client.listExtensions()
  } finally { client.close() }
})
